### DISCUSSION TITLE
SUN calculation

### DISCUSSION THREAD

**Discussion by ffrisenn** [FeynCalc/feyncalc#174] [General]
Dear all,

I am using FeynCalc 9.3.1 and I need to compute some factors in color space. Going back to my calculations I guess I have a problem with labeling indices. 
Therefore if I consider q(a1)qbar(a2)>g(a3)g(a4) where a1 and a2 are color indices in the fundamental representation and a3 and a4 in the adjoint.

I need to do the compute the following:

c8 = SUND[a3, a4, c] . SUNTF[c, a1, a2]

ComplexConjugate[c8] * SUNTF[a3, a1, c] . 
  SUNTF[a4, c, a2] // SUNSimplify
or
SUNTrace[ComplexConjugate[c8]*SUNTF[a3, a1, c] . SUNTF[a4, c, a2]]// SUNSimplify

They both don't give me any result, it doesn't simply the expression.

Does anybody know how to compute them properly?

Looking forward to hearing any suggestions that could help solve the problem.

Best, Fausto

**Comment by vsht** [comment]
Hi,

this should do it

```
c8 = SUND[a3, a4, c] . SUNTF[c, a1, a2]
ComplexConjugate[c8] * SUNTF[a3, a1, c] . SUNTF[a4, c, a2] // SUNSimplify[#,Explicit->True]&
(*-1/2*((4 - CA^2)*CF) *)
```

Cheers,
Vladyslav

**Comment by ffrisenn** [comment]
Dear Vladyslav,


thanks a lot for your solution. Indeed it was just sufficient to write Explicit-> True.


However, I have another issue you may know how to work out.

ComplexConjugate[SUNTF[a2, a1, a3] . SUNTF[a4, a3, a1]]*
   SUND[c, a2, a4] . SUNTF[c, a3, a1] // 
  SUNSimplify[#, SUNNToCACF -> False, Explicit -> True] & // Simplify
which gives 0


ComplexConjugate[SUND[c, a2, a4] . SUNTF[c, a3, a1]]*
   SUNTF[a2, a1, a3] . SUNTF[a4, a3, a1] // 
  SUNSimplify[#, SUNNToCACF -> False, Explicit -> True] & // Simplify

this one doesn't give a result...

Do you know why? 

Best, Fausto

**Reply by vsht** [reply]
Dear Fausto, 

sorry for the late reply. This should take care of it:

```
ComplexConjugate[SUND[c, a2, a4] . SUNTF[c, a3, a1]]*
    SUNTF[a2, a1, a3] . SUNTF[a4, a3, a1] // 
   SUNSimplify[#, SUNNToCACF -> False, Explicit -> True] & //  ReplaceAll[#, SUNTrace[x_, OptionsPattern[]] :> 
   SUNTrace[x, Explicit -> True]] & // Simplify
(*SUND[SUNIndex[a2], SUNIndex[a4], 
  SUNIndex[$AL[$67]]] SUNTF[{SUNIndex[a2], SUNIndex[a4], 
   SUNIndex[$AL[$67]]}, SUNFIndex[a1], SUNFIndex[a3]]*)
```

Cheers,
Vladyslav

**Comment by ffrisenn** [comment]
Dear Vladyslav,

Thanks for your answer!
> ComplexConjugate[SUND[c, a2, a4] . SUNTF[c, a3, a1]]*
>     SUNTF[a2, a1, a3] . SUNTF[a4, a3, a1] // 
>    SUNSimplify[#, SUNNToCACF -> False, Explicit -> True] & //  ReplaceAll[#, SUNTrace[x_, OptionsPattern[]] :> 
>    SUNTrace[x, Explicit -> True]] & // Simplify
> (*SUND[SUNIndex[a2], SUNIndex[a4], 
>   SUNIndex[$AL[$67]]] SUNTF[{SUNIndex[a2], SUNIndex[a4], 
>    SUNIndex[$AL[$67]]}, SUNFIndex[a1], SUNFIndex[a3]]*)
I guess the commented line is the result you get from the procedure you wrote. At least that’s the one I get. However, to my understanding, it should print 0.

Do you know why it doesn’t?

Best, Fausto

**Reply by vsht** [reply]
Dear Fausto, 

sorry, I was obviously too eager to share the code without having a closer look at your input expression first.

The indices must be wrong here: you have a3 and a1 appearing 3 times in 
```
ComplexConjugate[SUND[c, a2, a4] . SUNTF[c, a3, a1]]*SUNTF[a2, a1, a3] . SUNTF[a4, a3, a1]
```
which violates the Einstein summation convention. You can also see it from
```
ComplexConjugate[SUND[c, a2, a4] . SUNTF[c, a3, a1]]*
  SUNTF[a2, a1, a3] . SUNTF[a4, a3, a1] // FCCheckSyntax
```
Assuming that the proper expression should be
```
ComplexConjugate[SUND[c, a2, a4] . SUNTF[c, a33, a11]]*SUNTF[a2, a1, a3] . SUNTF[a4, a3, a1]
```
then `SUNSimplify` alone should give zero right away.

Cheers,
Vladyslav
